portfolio/src/main/java/com/google/sps/servlets/PlaylistServlet.java

```diff
@@ -0,0 +1,121 @@
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.gson.Gson;
+
+import com.google.sps.data.PropertyNames;
+import com.google.sps.data.Video;
+
+@WebServlet("/playlist")
+public class PlaylistServlet extends HttpServlet{
+
+    private static DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    private static boolean DEBUG = false;
+
+    // Build youtube service using dummy HttpRequestInitializer.
+    private static YouTube youtubeService = new YouTube.Builder(new NetHttpTransport(), new JacksonFactory(),
+        new HttpRequestInitializer() {
+            public void initialize(HttpRequest request) throws IOException {
+            }
+        })
+        .setApplicationName("playlist-items").build();
+
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException{
+
+        String playlistId = request.getParameter(PropertyNames.PLAYLIST_ID);
+        String apiKey = (String) datastore.prepare(new Query("YoutubeAPIKey"))
+            .asSingleEntity()
+            .getProperty("Key");
+
+        YouTube.PlaylistItems.List playlistRequest = youtubeService
+            .playlistItems()
+            .list("snippet,status")
+            .setFields("items("+
+                "snippet/title," +
+                "snippet/description," +
+                "snippet/thumbnails/default/url," +
+                "snippet/resourceId/videoId," +
+                "status/privacyStatus)," +
+                "nextPageToken");
+        List<Video> playlistVideos = new ArrayList<>();
+        PlaylistItemListResponse playlistResponse;
+        // Loop through all pages of results until there are none left.
+        do {
+            playlistResponse = playlistRequest
+            .setKey(apiKey)
+            .setPlaylistId(playlistId)
+            .setMaxResults(50L)
+            .execute();
+            for (PlaylistItem item : playlistResponse.getItems()) {
+                playlistVideos.add(Video.videoFromPlaylistItem(item));
+            }
+            if (playlistResponse.getNextPageToken() != null) {
+                playlistRequest = playlistRequest.setPageToken(playlistResponse.getNextPageToken());
+            }
+        } while (playlistResponse.getNextPageToken() != null);
+        // Add all videos and dummy analyses to datastore.
+        for (Video video: playlistVideos) {
+            Entity videoEntity = Video.videoToDatastoreEntity(video);
+            datastore.put(videoEntity);
+            // Create an analysis entity that is a child of the video entity with id 1.
+            Entity analysisEntity = new Entity("Analysis", 1L, videoEntity.getKey());
```

Comment:
Same comment as above: leave the parameter names as comments.

Reply:
Fixed.